tests/test_md.py

```diff
@@ -630,3 +646,114 @@ def test_heating_md(tmp_path):
     assert stat_data.units[0] == ""
     assert stat_data.units[16] == "K"
     assert stat_data.labels[16] == "Target T"
+
+
+def test_heating_files():
+    """Test default heating file names."""
+    traj_heating_path = Path("Cl4Na4-nvt-T10-T20-traj.xyz")
+    stats_heating_path = Path("Cl4Na4-nvt-T10-T20-stats.dat")
+    final_10_path = Path("Cl4Na4-nvt-T10-final.xyz")
+    final_20_path = Path("Cl4Na4-nvt-T20-final.xyz")
+
+    assert not traj_heating_path.exists()
+    assert not stats_heating_path.exists()
+    assert not final_10_path.exists()
+    assert not final_20_path.exists()
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    nvt = NVT(
+        struct=single_point.struct,
+        temp=25.0,
+        steps=0,
+        traj_every=2,
+        stats_every=2,
+        temp_start=10,
+        temp_end=20,
+        temp_step=10,
+        temp_time=2,
+    )
+    try:
+        nvt.run()
+        final_10_atoms = read(final_10_path)
+        assert isinstance(final_10_atoms, Atoms)
+        final_20_atoms = read(final_20_path)
+        assert isinstance(final_20_atoms, Atoms)
+
+        traj = read(traj_heating_path, index=":")
+        assert all(isinstance(image, Atoms) for image in traj)
+        assert len(traj) == 3
+
+        stats = Stats(source=stats_heating_path)
+        assert stats.rows == 3
+        assert stats.data[0, 16] == 10.0
+        assert stats.data[1, 16] == 10.0
+        assert stats.data[2, 16] == 20.0
+
+    finally:
+        traj_heating_path.unlink(missing_ok=True)
+        stats_heating_path.unlink(missing_ok=True)
+        final_10_path.unlink(missing_ok=True)
+        final_20_path.unlink(missing_ok=True)
+
+
+def test_heating_md_files():
+    """Test default heating files when also running md"""
+
+    traj_heating_path = Path("Cl4Na4-nvt-T10-T20-T25.0-traj.xyz")
+    stats_heating_path = Path("Cl4Na4-nvt-T10-T20-T25.0-stats.dat")
+    final_10_path = Path("Cl4Na4-nvt-T10-final.xyz")
+    final_20_path = Path("Cl4Na4-nvt-T20-final.xyz")
+    final_path = Path("Cl4Na4-nvt-T25.0-final.xyz")
+
+    assert not traj_heating_path.exists()
+    assert not stats_heating_path.exists()
+    assert not final_10_path.exists()
+    assert not final_20_path.exists()
+    assert not final_path.exists()
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    nvt = NVT(
+        struct=single_point.struct,
+        temp=25.0,
+        steps=100,
```

Comment:
I think we might want to reduce this and/or look at other options, as this test takes significantly longer to run than any others, and I'm not sure we gain much by having a large number of additional steps. E.g. the slowest currently:

```
11.47s call     tests/test_md.py::test_heating_md_files
2.16s call     tests/test_md.py::test_npt
2.15s call     tests/test_md.py::test_nve
2.06s call     tests/test_md.py::test_minimize_every
1.68s call     tests/test_md.py::test_init[NVT-nvt]
1.66s call     tests/test_md.py::test_nph
```

Reply:
Hmm I did just run this on scarf and obtained, I don't think I did anything to change the performance characteristics, but I'm now seeing ```test_npt``` as the slowest

```shell
pytest -v tests/test_md.py --durations=0
```

```
============================================================================================ slowest durations =============================================================================================
15.80s call     tests/test_md.py::test_init[NVT-nvt]
10.78s call     tests/test_md.py::test_npt
7.05s call     tests/test_md.py::test_atoms_struct
4.57s call     tests/test_md.py::test_heating_md_files
0.70s call     tests/test_md.py::test_minimize_every
0.64s call     tests/test_md.py::test_restart
0.63s call     tests/test_md.py::test_heating_md
0.49s call     tests/test_md.py::test_traj_start
0.43s call     tests/test_md.py::test_minimize
0.43s call     tests/test_md.py::test_heating_files
0.41s call     tests/test_md.py::test_nvt_nh
0.39s call     tests/test_md.py::test_nph
0.38s call     tests/test_md.py::test_nve
0.38s call     tests/test_md.py::test_rescale_every
0.36s call     tests/test_md.py::test_rotate_restart
0.21s call     tests/test_md.py::test_heating
0.16s call     tests/test_md.py::test_init[NVE-nve]
0.13s call     tests/test_md.py::test_remove_rot
0.11s call     tests/test_md.py::test_init[NPT-npt]
0.11s call     tests/test_md.py::test_init[NPH-nph]
0.10s call     tests/test_md.py::test_init[NVT_NH-nvt-nh]
0.08s call     tests/test_md.py::test_reset_velocities
0.07s call     tests/test_md.py::test_noramp_heating
0.04s setup    tests/test_md.py::test_nve
```

Reply:
scarf can be moody on io so i suspect you see the variation based on that.

Reply:
Testing locally, `test_heating_md_files`, is fairly consistently the slowest for me by a noticeable amount.

However, I think the biggest thing isn't the relative speed, but more whether we gain anything from doing say 100 steps rather than a smaller number during MD, when the test is just about checking it does MD after heating, not testing the MD itself as such (and similarly for how long we spend heating).

For reference, I think the largest number of MD steps we do elsewhere is 6.

Reply:
let us look at real time per step, the test itself will be slower since I expect we do few T. for unit testing small steps number eg 5 per T is ok... we shall have examples with more steps that we run manually/tag that are more science relevant

Reply:
There should be a reduced load now, still covering the 10, 20, and 25 (MD) temperatures

```
7.03s call     tests/test_md.py::test_atoms_struct
4.03s call     tests/test_md.py::test_npt
2.04s call     tests/test_md.py::test_init[NVT-nvt]
0.74s call     tests/test_md.py::test_minimize_every
0.67s call     tests/test_md.py::test_heating_md
0.56s call     tests/test_md.py::test_heating_md_files
0.56s call     tests/test_md.py::test_restart
0.51s call     tests/test_md.py::test_heating_files
0.50s call     tests/test_md.py::test_nve
0.48s call     tests/test_md.py::test_traj_start
0.41s call     tests/test_md.py::test_rescale_every
0.39s call     tests/test_md.py::test_minimize
0.39s call     tests/test_md.py::test_nph
0.38s call     tests/test_md.py::test_rotate_restart
0.37s call     tests/test_md.py::test_nvt_nh
0.28s call     tests/test_md.py::test_heating
0.14s call     tests/test_md.py::test_init[NVE-nve]
0.11s call     tests/test_md.py::test_init[NPT-npt]
0.11s call     tests/test_md.py::test_remove_rot
0.10s call     tests/test_md.py::test_init[NPH-nph]
0.10s call     tests/test_md.py::test_init[NVT_NH-nvt-nh]
0.08s call     tests/test_md.py::test_noramp_heating
0.08s call     tests/test_md.py::test_reset_velocities
0.03s setup    tests/test_md.py::test_nve
```

Reply:
That seems more reasonable, thanks!